anta/tests/routing/bgp.py

```diff
@@ -277,3 +362,323 @@ def test(self) -> None:
+
+        for command in self.instance_commands:
+            if command.params:
+                peer_count = 0
+                command_output = command.json_output
+
+                afi = cast(Afi, command.params.get("afi"))
+                safi = cast(Optional[Safi], command.params.get("safi"))
+                afi_vrf = cast(str, command.params.get("vrf"))
+                num_peers = cast(PositiveInt, command.params.get("num_peers"))
+
+                if not (vrfs := command_output.get("vrfs")):
+                    _add_bgp_failures(failures=failures, afi=afi, safi=safi, vrf=afi_vrf, issue="Not Configured")
+                    continue
+
+                if afi_vrf == "all":
+                    for vrf_data in vrfs.values():
+                        peer_count += len(vrf_data["peers"])
+                else:
+                    peer_count += len(command_output["vrfs"][afi_vrf]["peers"])
+
+                if peer_count != num_peers:
+                    _add_bgp_failures(failures=failures, afi=afi, safi=safi, vrf=afi_vrf, issue=f"Expected: {num_peers}, Actual: {peer_count}")
+
+        if failures:
+            self.result.is_failure(f"Failures: {list(failures.values())}")
+
+
+class VerifyBGPPeersHealth(AntaTest):
+    """
+    This test verifies the health of BGP peers.
+
+    It will validate that all BGP sessions are established and all message queues for these BGP sessions are empty for a given address family.
+
+    It supports multiple types of address families (AFI) and subsequent service families (SAFI).
+    Please refer to the Input class attributes below for details.
+
+    Expected Results:
+        * success: If all BGP sessions are established and all messages queues are empty for each address family and VRF.
+        * failure: If there are issues with any of the BGP sessions, or if BGP is not configured for an expected VRF or address family.
+    """
+
+    name = "VerifyBGPPeersHealth"
+    description = "Verifies the health of BGP peers"
+    categories = ["routing", "bgp"]
+    commands = [
+        AntaTemplate(template="show bgp {afi} {safi} summary vrf {vrf}"),
+        AntaTemplate(template="show bgp {afi} summary"),
+    ]
+
+    class Input(AntaTest.Input):  # pylint: disable=missing-class-docstring
+        address_families: List[BgpAfi]
+        """
+        List of BGP address families (BgpAfi)
+        """
+
+        class BgpAfi(BaseModel):  # pylint: disable=missing-class-docstring
+            afi: Afi
+            """BGP address family (AFI)"""
+            safi: Optional[Safi] = None
+            """Optional BGP subsequent service family (SAFI).
+
+            If the input `afi` is `ipv4` or `ipv6`, a valid `safi` must be provided.
+            """
+            vrf: str = "default"
+            """
+            Optional VRF for IPv4 and IPv6. If not provided, it defaults to `default`.
+
+            If the input `afi` is not `ipv4` or `ipv6`, e.g. `evpn`, `vrf` must be `default`.
+            """
```

Comment:
If this is the same across tests we should make this a class external to any test. I am thinking the "number of peers" can be added somehow by subclassing the main class to avoid repeating code